src/TopDownProteomics/ProForma/ProFormaParser.cs

```diff
@@ -156,6 +156,10 @@ public ProFormaTerm ParseString(ReadOnlySpan<char> proFormaString)
 
                         //i++; // skip the ? character
                     }
+                    else if (sequence.Length == 0 && proFormaString[i + 1] != '-')
+                    {
+                        throw new ProFormaParseException($"- must follow n terminal descriptor. - at index {i} is required.");
```

Comment:
There are a couple options that one can use at the beginning of a ProForma string, with the "-" being one of them. Does the logic work if the conditional was only `else if (sequence.Length == 0)`? Also, please adjust the exception language to something more generic like "invalid n terminal descriptor"